src/logging.rs

```diff
@@ -97,6 +105,47 @@ pub mod trace {
         }
     }
 
+    impl LevelHandle {
+        /// Returns a new `LevelHandle` without a corresponding filter.
+        ///
+        /// This will do nothing, but is required for admin endpoint tests which
+        /// do not exercise the `proxy-log-level` endpoint.
+        pub fn dangling() -> Self {
+            let builder = subscriber_builder()
+                .with_filter(filter::EnvFilter::default())
+                .with_filter_reloading();
+            let inner = builder.reload_handle();
+            LevelHandle { inner }
+        }
+
+        pub fn set_level(&self, level: impl AsRef<str>) -> Result<(), Error> {
+            let level = level.as_ref();
+            let filter = level.parse::<filter::EnvFilter>()?;
+            self.inner.reload(filter)?;
+            info!(message = "set new log level", %level);
+            Ok(())
+        }
+
+        pub fn current(&self) -> Result<String, Error> {
+            self.inner
+                .with_current(|f| format!("{}", f))
+                .map_err(Into::into)
+        }
+    }
+
+    impl fmt::Debug for LevelHandle {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            let current = self
+                .inner
+                // ugh...
```

Comment:
???

Reply:
ah, i had wanted to get rid of the unnecessary allocation from `format!` but i forgot. will fix it!

Reply:
fixed in ccca5a